src/components/category-selection/category-badge/CategoryBadge.tsx

```diff
@@ -1,31 +1,46 @@
 import classes from './CategoryBadge.module.scss';
 
 import { MouseEvent, TouchEvent, useState } from 'react';
-import CategoryModal from '../shared/CategoryModal';
-import { Category } from '../../data/interfaces';
+import CategoryModal from '../../shared/CategoryModal';
+import { Category } from '../../../data/interfaces';
 import classNames from 'classnames/bind';
 
 const cx = classNames.bind(classes);
 
-const CategoryBadge = ({ id, name, color }: Category) => {
+export interface CategoryBadgeProps extends Category {
+    onClick?: () => void;
+    deletable?: boolean;
+}
+
+const CategoryBadge = ({ 
+    id, 
+    name, 
+    color, 
+    onClick, 
+    deletable 
+}: CategoryBadgeProps) => {
     const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
 
     const handleOnClick = (event: MouseEvent | TouchEvent) => {
         event.stopPropagation();
-        setIsModalOpen(true);
+        // TODO remove monkey patch
```

Comment:
For "TODO" comments, please utilize our project board instead, to keep track of pending tasks and ensure a clutter-free codebase.

Reply:
Why not just do it right away? Monkey patches are no good for prod anyway

Reply:
Actually I took a deeper look now, we should give it more thought later. Will add a task to the backlog.